### DISCUSSION TITLE
Generic CRUD

### DISCUSSION THREAD

**Discussion by santo998** [linq2db/linq2db#3101] [Q&A] [answered]
1) How can i access to id property in a generic way? i mean, if i dont know the model type at build time.

i.e.: if i have this two models:
class Client { public int Id; public string Name; }
class Invoice { public int Id; }

And then i have a generic method:
void PrintId<TModel>(TModel model)
{
// TO DO: print model Id based on runtime type
}

2) Also how can i access to all the model properties?

void PrintProperties<TModel>(TModel model)
{
// TO DO: print model properties based on runtime type
// should print Id for Invoice, then Id and Name for Client object
}

On the methods, add the where clauses that you consider helpful. Maybe where TModel : ITable<TModel> or something.
Also i would like to avoid reflection if possible.
I wonder if MappingSchema or some ITable property can help me...

**Comment by sdanyliv** [comment, marked as answer]
Try `MappingSchema.GetEntityDescriptor(Type)`
